src/app/main.py

```diff
@@ -23,12 +23,24 @@ def new_dataset(cloud_event):
     bucket_name = cloud_event.data["bucket"]
     filename = cloud_event.data["name"]
 
-    dataset = dataset_storage.get_dataset(filename=filename, bucket_name=bucket_name)
-
-    logger.info("Dataset obtained successfully.")
-    logger.debug(f"Dataset: {dataset}")
-
-    dataset_id = str(uuid.uuid4())
-    database.set_dataset(dataset_id=dataset_id, filename=filename, dataset=dataset)
-
-    logger.info("Dataset uploaded successfully.")
+    """
+    Check if the filename ends with '.json'.
+    Process the contents if it is a valid '.json' file or else log the error.
+    """
+    if filename[-5:].lower() == ".json":
+        dataset = dataset_storage.get_dataset(
+            filename=filename, bucket_name=bucket_name
+        )
+
+        if dataset is not None:
+            logger.info("Dataset obtained successfully.")
+            logger.debug(f"Dataset: {dataset}")
+            dataset_id = str(uuid.uuid4())
+            database.set_dataset(
+                dataset_id=dataset_id, filename=filename, dataset=dataset
+            )
+            logger.info("Dataset uploaded successfully.")
+        else:
+            logger.error("Invalid JSON file contents")
+    else:
+        logger.error(f"Invalid filetype received - {filename}")
```

Comment:
Please can you add tests for the different conditions happening here?
Tests for invalid json file names
Test for when dataset is none

Reply:
@JackRimmerAND Thanks a lot for this review, just added the logging tests as two new unit tests in 'new_dataset_logs_test.py' and another test in 'dataset_storage_test.py' specifically for the 'validate_keys' method.
I've added a pytest fixture in conftest.py for the unit test in dataset_storage_test, hope this looks okay now. I see that the test coverage has improved considerably and is now 100% for main.py and 90% for dataset_storage.py.